### PR TITLE
[WIP] upgrade to django-oscar 2.1

### PR DESCRIPTION
**Pull request by dianekaplan**
To support moving to django 3.2, we need to first upgrade our django-oscar package to 2.1 

This work has involved a series of changes just to make the upgrade possible: 
- clearing the path to get 'make upgrade' working again (REV-2369)
- deploy pending minor package version updates ([here](https://github.com/edx/ecommerce/pull/3512))
- update the way we do coupon-category-creating migration scripts, since our previous approach clashes with any oscar upgrade that changes that model (like the one for 2.1 will do)- REV-2416
- pin bok-choy, which won't support the next selenium upgrade that 'make upgrade' will try (REV-2451)
- remove karma-jquery, since it won't support jquery 3.5 (needed by django-oscar 2.1)- REV-2453

This PR has the changes needed to accommodate django-oscar 2.1, and the actual version upgrade! 

Testing notes: the work listed above has gotten us to the point of having these build checks passing, which they finally do (minus slight codecov %age drop). The larger testing effort is being planned in this request: https://openedx.atlassian.net/browse/CR-4129

### CODE REVIEW

File: ecommerce/extensions/catalogue/migrations/0013_coupon_product_class.py [hunk 1]
```diff
@@ -2,19 +2,18 @@
...
-
 
 def create_product_class(apps, schema_editor):
     """Create a Coupon product class."""
-    for klass in (Category, ProductAttribute, ProductClass):
+    Category = apps.get_model("catalogue", "Category")
+    ProductAttribute = apps.get_model("catalogue", "ProductAttribute")
+    ProductClass = apps.get_model("catalogue", "ProductClass")
+
+    for klass in (ProductAttribute, ProductClass):
```
Comment (inventhouse) [inline ecommerce/extensions/catalogue/migrations/0013_coupon_product_class.py:16]: Just checking that you are intentionally no longer setting `skip_history_when_saving` on `Category`, correct?

File: ecommerce/extensions/catalogue/migrations/0025_course_entitlement.py [hunk 2]
```diff
@@ -1,20 +1,18 @@
...
-ProductClass = get_model("catalogue", "ProductClass")
-
 
 def create_product_class(apps, schema_editor):
     """ Create a course entitlement product class """
+    Category = apps.get_model("catalogue", "Category")
+    ProductAttribute = apps.get_model("catalogue", "ProductAttribute")
+    ProductClass = apps.get_model("catalogue", "ProductClass")
+
     for klass in (Category, ProductAttribute, ProductClass):
```
Comment (inventhouse) [inline ecommerce/extensions/catalogue/migrations/0025_course_entitlement.py:16]: I note that here you (or `makemigrations`) are continuing to set `skip_history_when_saving` on `Category`, where you don't above; if that's intentional, perhaps a comment in one place or the other would be helpful.  (I also see a few more places both with and without setting this below)

File: karma.conf.js [hunk 3]
```diff
@@ -21,7 +21,7 @@ module.exports = function(config) {
     // available frameworks: https://npmjs.org/browse/keyword/karma-adapter
     frameworks: ['jasmine', 'requirejs', 'sinon'],
 
-
+
```
Comment (inventhouse) [inline karma.conf.js:24]: Whitespace change that can probably be reverted

**Comment by dianekaplan** [comment]
Closing in favor of this cleaner one here: https://github.com/edx/ecommerce/pull/3572/commits

**Pull request closed without merging**